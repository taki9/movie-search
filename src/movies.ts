export default {
  count: 5042,
  next: 'http://imdb.hriks.com/movie/?limit=50&offset=50',
  previous: null,
  results: [
    {
      id: 2844,
      title: 'Justin Bieber: Never Say Never',
      date_of_release: '2011-01-01',
      rating: 1.6,
      tags: ['documentary', 'music'],
      director: '7745: Jon M. Chu',
    },
    {
      id: 1146,
      title: 'Foodfight! ',
      date_of_release: '2012-01-01',
      rating: 1.7,
      tags: ['action', 'fantasy', 'animation', 'comedy', 'family'],
      director: '8202: Lawrence Kasanoff',
    },
    {
      id: 2278,
      title: 'Disaster Movie ',
      date_of_release: '2008-01-01',
      rating: 1.9,
      tags: ['comedy'],
      director: '8418: Jason Friedberg',
    },
    {
      id: 2305,
      title: 'Superbabies: Baby Geniuses 2 ',
      date_of_release: '2004-01-01',
      rating: 1.9,
      tags: ['sci-fi', 'comedy', 'family'],
      director: '8701: Bob Clark',
    },
    {
      id: 4615,
      title: 'The Helix... Loaded ',
      date_of_release: '2005-01-01',
      rating: 1.9,
      tags: ['action', 'adventure', 'fantasy', 'sci-fi', 'comedy'],
      director: '9978: A. Raven Cruz',
    },
    {
      id: 1739,
      title: 'United Passions ',
      date_of_release: '2014-01-01',
      rating: 2,
      tags: ['drama', 'history', 'sport'],
      director: '8463: Frédéric Auburtin',
    },
    {
      id: 3515,
      title: "Who's Your Caddy? ",
      date_of_release: '2007-01-01',
      rating: 2,
      tags: ['comedy', 'sport'],
      director: '9341: Don Michael Paul',
    },
    {
      id: 2993,
      title: 'From Justin to Kelly ',
      date_of_release: '2003-01-01',
      rating: 2.1,
      tags: ['romance', 'comedy', 'musical'],
      director: '8838: Robert Iscove',
    },
    {
      id: 3350,
      title: 'Glitter ',
      date_of_release: '2001-01-01',
      rating: 2.1,
      tags: ['romance', 'drama', 'music'],
      director: '3775: Vondie Curtis-Hall',
    },
    {
      id: 3674,
      title: 'Crossover ',
      date_of_release: '2006-01-01',
      rating: 2.1,
      tags: ['action', 'sport'],
      director: '8991: Preston A. Whitmore II',
    },
    {
      id: 329,
      title: 'Son of the Mask ',
      date_of_release: '2005-01-01',
      rating: 2.2,
      tags: ['fantasy', 'comedy', 'family'],
      director: '7826: Lawrence Guterman',
    },
    {
      id: 4629,
      title: 'Snow White: A Deadly Summer ',
      date_of_release: '2012-01-01',
      rating: 2.2,
      tags: ['horror'],
      director: '9988: David DeCoteau',
    },
    {
      id: 4778,
      title: 'Subconscious ',
      date_of_release: '2015-01-01',
      rating: 2.2,
      tags: ['action', 'thriller', 'mystery'],
      director: '10086: Georgia Hilton',
    },
    {
      id: 2202,
      title: 'Epic Movie ',
      date_of_release: '2007-01-01',
      rating: 2.3,
      tags: ['adventure', 'comedy'],
      director: '8418: Jason Friedberg',
    },
    {
      id: 2323,
      title: 'Alone in the Dark ',
      date_of_release: '2005-01-01',
      rating: 2.3,
      tags: ['sci-fi', 'horror'],
      director: '8710: Uwe Boll',
    },
    {
      id: 3535,
      title: 'Chairman of the Board ',
      date_of_release: '1998-01-01',
      rating: 2.3,
      tags: ['comedy'],
      director: '9353: Alex Zamm',
    },
    {
      id: 505,
      title: 'Battlefield Earth ',
      date_of_release: '2000-01-01',
      rating: 2.4,
      tags: ['action', 'adventure', 'sci-fi'],
      director: '7904: Roger Christian',
    },
    {
      id: 909,
      title: 'Gigli ',
      date_of_release: '2003-01-01',
      rating: 2.4,
      tags: ['romance', 'comedy', 'crime'],
      director: '7870: Martin Brest',
    },
    {
      id: 2862,
      title: 'Baby Geniuses ',
      date_of_release: '1999-01-01',
      rating: 2.5,
      tags: ['sci-fi', 'comedy', 'family', 'crime'],
      director: '8992: Bob Clark',
    },
    {
      id: 4884,
      title: 'Sunday School Musical ',
      date_of_release: '2008-01-01',
      rating: 2.5,
      tags: ['musical', 'drama'],
      director: '10149: Rachel Goldenberg',
    },
    {
      id: 4143,
      title: '30 Nights of Paranormal Activity with the Devil Inside the Girl with the Dragon Tattoo ',
      date_of_release: '2013-01-01',
      rating: 2.6,
      tags: ['comedy'],
      director: '9705: Craig Moss',
    },
    {
      id: 4535,
      title: 'The Vatican Exorcisms ',
      date_of_release: '2013-01-01',
      rating: 2.6,
      tags: ['horror'],
      director: '6666: Joe Marino',
    },
    {
      id: 1082,
      title: 'Inchon ',
      date_of_release: '1981-01-01',
      rating: 2.7,
      tags: ['drama', 'history', 'war'],
      director: '8175: Terence Young',
    },
    {
      id: 1664,
      title: 'Meet the Spartans ',
      date_of_release: '2008-01-01',
      rating: 2.7,
      tags: ['comedy'],
      director: '8418: Jason Friedberg',
    },
    {
      id: 1712,
      title: 'Dragonball: Evolution ',
      date_of_release: '2009-01-01',
      rating: 2.7,
      tags: ['action', 'adventure', 'fantasy', 'sci-fi', 'thriller'],
      director: '8161: James Wong',
    },
    {
      id: 2192,
      title: 'Date Movie ',
      date_of_release: '2006-01-01',
      rating: 2.7,
      tags: ['romance', 'comedy'],
      director: '8650: Aaron Seltzer',
    },
    {
      id: 3933,
      title: 'The Real Cancun ',
      date_of_release: '2003-01-01',
      rating: 2.7,
      tags: ['documentary'],
      director: '9578: Rick de Oliveira',
    },
    {
      id: 4482,
      title: 'The 41-Year-Old Virgin Who Knocked Up Sarah Marshall and Felt Superbad About It ',
      date_of_release: '2010-01-01',
      rating: 2.7,
      tags: ['comedy'],
      director: '9705: Craig Moss',
    },
    {
      id: 2076,
      title: 'Jaws: The Revenge ',
      date_of_release: '1987-01-01',
      rating: 2.8,
      tags: ['adventure', 'thriller', 'horror'],
      director: '8607: Joseph Sargent',
    },
    {
      id: 2311,
      title: 'Doogal ',
      date_of_release: '2006-01-01',
      rating: 2.8,
      tags: ['adventure', 'fantasy', 'animation', 'comedy', 'family'],
      director: '8703: Dave Borthwick',
    },
    {
      id: 2338,
      title: 'Marci X ',
      date_of_release: '2003-01-01',
      rating: 2.8,
      tags: ['comedy', 'music'],
      director: '8576: Richard Benjamin',
    },
    {
      id: 2591,
      title: 'Steel ',
      date_of_release: '1997-01-01',
      rating: 2.8,
      tags: ['action', 'sci-fi', 'crime'],
      director: '8848: Kenneth Johnson',
    },
    {
      id: 2706,
      title: "Barney's Great Adventure ",
      date_of_release: '1998-01-01',
      rating: 2.8,
      tags: ['adventure', 'family'],
      director: '8907: Steve Gomer',
    },
    {
      id: 3028,
      title: "Dancin' It's On ",
      date_of_release: '2015-01-01',
      rating: 2.8,
      tags: ['family', 'musical'],
      director: '9079: David Winters',
    },
    {
      id: 4309,
      title: 'Hum To Mohabbat Karega ',
      date_of_release: '2000-01-01',
      rating: 2.8,
      tags: ['action', 'thriller', 'romance', 'comedy'],
      director: '9798: Kundan Shah',
    },
    {
      id: 4453,
      title: 'Mutant World ',
      date_of_release: '2014-01-01',
      rating: 2.8,
      tags: ['sci-fi'],
      director: '9885: David Winning',
    },
    {
      id: 4623,
      title: 'Frat Party ',
      date_of_release: '2009-01-01',
      rating: 2.8,
      tags: ['comedy'],
      director: '9985: Robert Bennett',
    },
    {
      id: 2008,
      title: 'BloodRayne ',
      date_of_release: '2005-01-01',
      rating: 2.9,
      tags: ['action', 'adventure', 'fantasy', 'horror'],
      director: '8585: Uwe Boll',
    },
    {
      id: 2019,
      title: "The True Story of Puss'N Boots ",
      date_of_release: '2009-01-01',
      rating: 2.9,
      tags: ['adventure', 'animation', 'comedy'],
      director: '3127: Jérôme Deschamps',
    },
    {
      id: 3610,
      title: 'Daddy Day Camp ',
      date_of_release: '2007-01-01',
      rating: 2.9,
      tags: ['comedy', 'family'],
      director: '9398: Fred Savage',
    },
    {
      id: 630,
      title: 'Rollerball ',
      date_of_release: '2002-01-01',
      rating: 3,
      tags: ['action', 'sci-fi', 'sport'],
      director: '7771: John McTiernan',
    },
    {
      id: 2532,
      title: 'Shanghai Surprise ',
      date_of_release: '1986-01-01',
      rating: 3,
      tags: ['adventure', 'romance', 'drama', 'crime'],
      director: '8816: Jim Goddard',
    },
    {
      id: 4091,
      title: 'Phat Girlz ',
      date_of_release: '2006-01-01',
      rating: 3,
      tags: ['comedy'],
      director: '9673: Nnegest Likké',
    },
    {
      id: 4534,
      title: 'The Dead Undead ',
      date_of_release: '2010-01-01',
      rating: 3,
      tags: ['action', 'sci-fi', 'horror'],
      director: '6663: Matthew R. Anderson',
    },
    {
      id: 5030,
      title: 'The Ridges ',
      date_of_release: '2011-01-01',
      rating: 3,
      tags: ['thriller', 'drama', 'horror'],
      director: '7586: Brandon Landers',
    },
    {
      id: 2409,
      title: 'Left Behind ',
      date_of_release: '2014-01-01',
      rating: 3.1,
      tags: ['action', 'fantasy', 'thriller', 'mystery', 'drama'],
      director: '8754: Vic Armstrong',
    },
    {
      id: 2521,
      title: 'Larry the Cable Guy: Health Inspector ',
      date_of_release: '2006-01-01',
      rating: 3.1,
      tags: ['romance', 'comedy'],
      director: '8808: Trent Cooper',
    },
    {
      id: 3455,
      title: 'Witless Protection ',
      date_of_release: '2008-01-01',
      rating: 3.1,
      tags: ['comedy', 'crime'],
      director: '9317: Charles Robert Carner',
    },
    {
      id: 4032,
      title: 'Lords of London ',
      date_of_release: '2014-01-01',
      rating: 3.1,
      tags: ['thriller', 'mystery', 'crime'],
      director: '9641: Antonio Simoncini',
    },
  ],
};
