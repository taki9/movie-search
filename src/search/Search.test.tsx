import { expect, describe, it, afterEach, vi } from 'vitest';
import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { Search } from '.';
import movies from '../movies';

const renderComponent = () => render(<Search onChange={vi.fn()} />);

describe('Search component', () => {
  afterEach(cleanup);

  it('should match the snapshot', () => {
    const { container } = renderComponent();

    expect(container.innerHTML).toMatchSnapshot();
  });

  it('should search for items', () => {
    const { getAllByTestId, getByTestId } = renderComponent();
    const term = 'gl';
    const expectedItems = movies.results.reduce<string[]>((acc, current) => {
      if (current.title.match(new RegExp(term, 'i'))) {
        acc.push(current.title);
      }
      return acc;
    }, []);

    const input = getByTestId('input-field');
    userEvent.type(input, 'gl');
    expect((input as HTMLInputElement).value).toBe('gl');

    const listItems = getAllByTestId('list-item');

    expect(listItems).toHaveLength(2);

    listItems.map((item) => {
      const title = item.querySelector('.MuiListItemText-primary');
      expect(title).toBeTruthy();
      expect(expectedItems).toContain(title!.textContent);
    });
  });
});
