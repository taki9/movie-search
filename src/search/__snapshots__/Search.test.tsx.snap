// Vitest Snapshot v1

exports[`Search component > should match the snapshot 1`] = `"<div class=\\"MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1u49069-MuiAutocomplete-root\\" role=\\"combobox\\" aria-expanded=\\"false\\"><div class=\\"MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root\\"><div class=\\"MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart MuiAutocomplete-inputRoot css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root\\"><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root\\"><span class=\\"notranslate\\">​</span><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"SearchIcon\\"><path d=\\"M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z\\"></path></svg></div><input aria-invalid=\\"false\\" autocomplete=\\"off\\" placeholder=\\"Search...\\" type=\\"text\\" class=\\"MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedStart MuiAutocomplete-input MuiAutocomplete-inputFocused css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input\\" aria-autocomplete=\\"list\\" autocapitalize=\\"none\\" spellcheck=\\"false\\" data-testid=\\"input-field\\" value=\\"\\" id=\\"mui-2\\"><fieldset aria-hidden=\\"true\\" class=\\"MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline\\"><legend class=\\"css-nnbavb\\"><span class=\\"notranslate\\">​</span></legend></fieldset></div></div></div>"`;
